src/data_server/s3service.rs

```diff
@@ -87,7 +148,7 @@ impl S3 for S3ServiceServer {
                 if !h.is_empty() && h.len() == 32 {
                     self.backend
                         .head_object(ArunaLocation {
-                            bucket: format!("{}-{}", &self.endpoint_id, &h[0..2]),
+                            bucket: format!("{}-{}", &self.aruna_internal, &h[0..2]),
```

Comment:
This is broken, the `endpoint_id` is intentional here !

The expected format for bucket is `<ENDPOINT_ULID>-hash[0..2]` this should be unique per dataproxy and not contain any aruna_internal urls etc.